### ISSUE TITLE
need better checking of pattern semantics

### ISSUE THREAD

**Issue by jmuhlich** [pysb/pysb#2]
Things like this need to error early, instead of getting passed to BNG where undefined things can happen:

  Observe('R', R(lr=1, rf=1))

(1's should be ANY)

**Comment by alubbock** [comment]
I think the semantics of using `1` or `ANY` are different here. `R(lr=1, rf=1)` indicates an internal bond between the `lr` and `rf` sites. `R(lr=ANY, rf=ANY)` means that `lr` and `rf` could be bound to anything - either each other, or some other monomer/complex.

**Comment by lh64** [comment]
Yeah, that's true. I actually wasn't sure that `R(lr!1, rf!1)` was valid BNGL syntax but I tested it and it is. It's kind of weird (I've never seen a model that creates internal bonds like this) but I suppose it could be useful.
